Router

atonkopi opened the pull request.

atonkopi committed:
Initial set up and render Subreddits

atonkopi committed:
Add default Subreddit pic if it fails to load

atonkopi committed:
Render Posts list

atonkopi committed:
Add Subreddits Search

atonkopi committed:
Posts list updates on click

atonkopi committed:
Fetch and render comments for each post

atonkopi committed:
Add replies to comments

atonkopi committed:
Implementation of pagination

atonkopi committed:
Fix pagination issue - click on subreddit update pagination

atonkopi committed:
Implement feature Search through Reddit

atonkopi committed:
Subreddits search

atonkopi committed:
Add style with MUI

atonkopi committed:
Add styling with MUI

atonkopi committed:
Add text and video if applicable to posts

atonkopi committed:
Separate post page implementation with React Router

id13 committed:
(update) Update routing to posts

- Fixed infinite rendering of post page
- Broke pagination (needs to be fixed)
- Post page does not work when it is loaded straight from the browser
instead of navigating from the Index page. (needs to be fixed too)
- Decreased the number of components relying on redux state

atonkopi committed:
Change color palette

atonkopi committed:
Merge branch 'router' of github.com:a-dylean/reddit-client-app into router

atonkopi committed:
Add single post page

- Updated Post Slice

atonkopi committed:
Fix video

- Eliminated ReactPlayer
- Added second MediaCard instead

atonkopi committed:
Add a new route

- Default page renders by default
- Home page renders on selected Subreddit click
- URL changes depending on selected subreddit

atonkopi committed:
Fix 1st page pagination bug

atonkopi committed:
Fixed routing bugs

- Added Layout component

atonkopi committed:
Fixed bug with URL on default page render

atonkopi committed:
Introduce changes to theme & Search component

atonkopi committed:
Changed routing

-Removed Default page and PostListsPage
-PageLayout deleted
-PostsListPage has default subreddit value and renders for both "/" and
selected subreddit

atonkopi committed:
Removed initial state of selected subreddit from the store

atonkopi committed:
Implement infinite loading with Infinite scroll component

atonkopi committed:
Improve post view (text coverage)

atonkopi committed:
Implement breakpoints for mobile version

atonkopi committed:
Improve mobile version (navbar)

atonkopi committed:
Improve toolbar

atonkopi committed:
Change position of toolbar components

- Changed position of Search component and burger menu

atonkopi committed:
Change color of Subreddits in mobile version

atonkopi committed:
Improve readability of text elements

- Changed font size in theme (reverted px into rem)
- Changed typography variants
- Adjusted width in comments

atonkopi committed:
Improvement of toolbar appearance

atonkopi committed:
Improve toolbar (mobile+desktop)

atonkopi committed:
Change data format in posts and comments

atonkopi committed:
Add new component Subreddit Info

- Subreddit info holds info about reddit community
Further improvements on subreddits list

atonkopi committed:
Fix bug with subreddit info

- Added a new action to get subreddit info to subreddits slice
- Fixed bugs in Subreddit info component
- Added new features (subscribers & active subscribers)

atonkopi committed:
Search results appear under search component

atonkopi committed:
Improve readability of the code

atonkopi committed:
Fixed bugs re searchbar and minor improvements

id13 commented on public/index.html:
```diff
@@ -2,42 +2,29 @@
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-    <title>React App</title>
+    <link
+  rel="stylesheet"
+  href="https://fonts.googleapis.com/icon?family=Material+Icons"/>
+  <link
+  rel="stylesheet"
+  href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
+/>
+<link rel="preconnect" href="https://fonts.googleapis.com">
+<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+<link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:wght@100;400;600&family=Noto+Sans:wght@400&display=swap" rel="stylesheet">
+    <title>Reddit Client App</title>
```
Could be indented

atonkopi replied:
Fixed

id13 commented on src/components/helperFunctions.js:
```diff
@@ -0,0 +1,49 @@
+      function handleResize() {
+        setWindowSize({
+          width: window.innerWidth,
+          height: window.innerHeight,
+        });
+      }
+      window.addEventListener("resize", handleResize);
+      handleResize();
+      return () => window.removeEventListener("resize", handleResize);
+    }
+  }, []);
+  return windowSize;
+};
+
+export const relativeDays = (timestamp) => {
+  const rtf = new Intl.RelativeTimeFormat("en", {
+    numeric: "auto",
+  });
+  const oneDayInHs = 1000 * 60 * 60;
+  const oneDayinMs = 1000 * 60 * 60 * 24;
+  const hoursDifference = Math.round(
+    (timestamp - new Date().getTime()) / oneDayInHs
+  );
+  const daysDifference = Math.round(
+    (timestamp - new Date().getTime()) / oneDayinMs
+  );
+  if (hoursDifference > -24) {
+    return rtf.format(hoursDifference, "hour")
+  } else {
+    return rtf.format(daysDifference, "day");
+  } 
+};
+
+export const numFormatter = (num) => {
+  return Math.abs(num) > 999999
+    ? Math.sign(num) * (Math.abs(num) / 1000000).toFixed(1) + "m"
+    : Math.abs(num) > 999
+    ? Math.sign(num) * (Math.abs(num) / 1000).toFixed(1) + "k"
+    : Math.sign(num) * Math.abs(num);
+};
```
I would put it an `helpers` folder outside of `components` like `src/helpers`, it's not realy related to components.
You can do it by creating the folder, then from VS Code drag and drop the file to the new location. 
This way, VS will automatically change the imports everywhere it was used to the new location

atonkopi replied:
Done

id13 commented on src/components/layout.js:
```diff
@@ -0,0 +1,41 @@
+        </Toolbar>
+      </AppBar>
+      <main>{children}</main>
+    </>
```
Several points about this code

1. Why having two functions that does low level document.getElementById of a html tag when you can just pass something like 
```
{displaySelectedSubreddit && <Typography
            id="selectedSubreddit"
            sx={{
              color: "white",
              cursor: "pointer",
            }}
            noWrap
            onClick={() => navigate(`/r/${selectedSubreddit}`)}
            variant="h3"
          >
            /r/{selectedSubreddit}
          </Typography>
}
```
or even pass a `sx={{display: displaySelectedSubreddit  ? 'block' : 'none' }}` (not sure it will work) ?
As for the displaySelectedSubreddit, it can be a `useState`, with the proper setter passed to `hideSudreddit` and `showSubreddit`.

2. About `hideSudreddit` and `showSubreddit`, the naming is too much coupled to the layout's logic. I would rather have had one props, called `onFocusChange` that passes the value `true` or `false` whether the search is focused. And then the layout handle the logic to display/hide selected subredit based on window size.

id13 commented on src/components/postPage.js:
```diff
@@ -0,0 +1,39 @@
+
+  return (
+    <Layout selectedSubreddit={selectedSubreddit}>
+      <Container sx={{ mt: "3rem" }}>
```
What's this for ?

id13 commented on src/components/postPage.js:
```diff
@@ -0,0 +1,39 @@
+            display: "flex",
+            justifyContent: "center",
+            alignItems: "flex-start",
+            pb: "1rem",
```
same question

atonkopi replied:
Removed unnecessary Box

id13 commented on src/components/postPage.js:
```diff
@@ -0,0 +1,39 @@
+          }}
+        ></Box>
+        {selectedPost ? (
+          <Post post={selectedPost} fullVersion={true} />
```
```suggestion
          <Post post={selectedPost} fullVersion />
```

id13 commented on src/components/postPage.js:
```diff
@@ -0,0 +1,39 @@
+        {selectedPost ? (
+          <Post post={selectedPost} fullVersion={true} />
+        ) : (
+          <Card><Typography variant="h7">Loading...</Typography><LinearProgress/></Card>
```
You probably want to have a loading state as a component

atonkopi replied:
Done

id13 commented on src/components/postsListPage.js:
```diff
@@ -0,0 +1,84 @@
+          justifyContent="center"
+          alignItems="flex-start"
+          wrap="nowrap"
+          spacing={3.5}
```
What's this ?

id13 commented on src/components/postsListPage.js:
```diff
@@ -0,0 +1,84 @@
+              ""
+            )}
+            <InfiniteScroll
+              dataLength={Object.values(posts).length}
```
```suggestion
              dataLength={Object.keys(posts).length}
```
Functionaly the same, but is probably faster and less memory intense

id13 commented on src/components/postsListPage.js:
```diff
@@ -0,0 +1,84 @@
+          <Grid item xs={16} md={8}>
+            {loading ? (
+              <Card>
+                <Typography variant="h7">Posts are loading...</Typography>
```
Same remark about loading states

atonkopi replied:
Done

id13 commented on src/components/postsListPage.js:
```diff
@@ -0,0 +1,84 @@
+                md={4}
+                sx={{
+                  width: "100%",
+                  top: "3rem",
+                  minWidth: "23rem",
```
what's this ?

atonkopi replied:
Removed unnecessary styling

id13 commented on src/components/postsListPage.js:
```diff
@@ -0,0 +1,84 @@
+                item
+                maxWidth="md"
+                xs={0}
+                md={4}
```
what's this ?

id13 commented on src/components/postsListPage.js:
```diff
@@ -0,0 +1,84 @@
+                  display: {xs: "none", sm: "block"} 
+                }}
+              >
+                <Card sx={{ mb: "1rem" }}>
```
What's this ?

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+const SearchDiv = styled("div")(({ theme }) => ({
+  borderRadius: 50,
+  backgroundColor: "#fafafa",
+  color: "rgba(0, 0, 0, 0.54)",
```
Should dereferenced from the injected theme, as the colours are global to the whole site, you don't want hardcoded values in some component. 

What if the designers change it ?

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+}));
+
+const SearchDiv = styled("div")(({ theme }) => ({
+  borderRadius: 50,
```
what's this ?

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: "black",
```
same thing about colors

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: "black",
+  "& .MuiInputBase-input": {
+    padding: theme.spacing(1, 1, 1, 0),
```
very good, anything related to margin or padding should be like that, it's more maintainable

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+  "& .MuiInputBase-input": {
+    padding: theme.spacing(1, 1, 1, 0),
+    [theme.breakpoints.down("sm")]: {
+      width: "0.1ch",
```
never seen this unit 👀

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+  dispatch(searchSubreddits(searchTerm));
+}, 300);
+
+export const Search = ({ hideSudreddit, showSubreddit }) => {
+  const size = useWindowSize();
+  const [searchTerm, setSearchTerm] = useState("");
+  const dispatch = useDispatch();
+
+  const searchTermChangeHandler = (e) => {
+    const newSearchTerm = e.target.value;
+    setSearchTerm(newSearchTerm);
+    searchSubredditsDebounced(dispatch, newSearchTerm);
+  };
+
+  const clearSearchTermHandler = () => {
+    setSearchTerm("");
+  };
+
+  const [showSubreddits, setShowSubreddits] = useState(false);
+
+  const handleMouseEnter = () => {
+    setShowSubreddits(true);
+  };
+
+  const handleMouseLeave = () => {
+    setShowSubreddits(false);
+  };
+
+  return (
+    <ParentDiv onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+      <SearchDiv onClick={(size.width < 400) ? hideSudreddit : undefined}>
```
The whole "show/hide selected subereddit" should be up to it's parent. Just fire a callback about focus change like told above

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+          value={searchTerm}
+          onChange={searchTermChangeHandler}
+          startAdornment={
+            <SearchIcon sx={{ ml: 2, mr: 1, color: "rgba(0, 0, 0, 0.54)" }} />
```
hardcoded values that should be dereferenced from the theme

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+        </StyledInputBase>
+        {searchTerm.length > 0 && (
+          <CancelOutlinedIcon
+            sx={{ mr: 2, color: "rgba(0, 0, 0, 0.54)" }}
```
same

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+      </SearchDiv>
+      <Card
+        sx={{
+          boxShadow: "rgba(0, 0, 0, 0.35) 0px 5px 15px",
```
same

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+      <Card
+        sx={{
+          boxShadow: "rgba(0, 0, 0, 0.35) 0px 5px 15px",
+          borderTopLeftRadius: 0,
+          borderTopRightRadius: 0,
```
same

id13 commented on src/components/search.js:
```diff
@@ -0,0 +1,114 @@
+          display: showSubreddits ? "block" : "none",
+        }}
+        onClick={() => {
+          setShowSubreddits(!showSubreddit);
+          showSubreddit();
```
Probably should be a function

id13 commented on src/components/theme.js:
```diff
@@ -0,0 +1,95 @@
+    },
+  },
+  palette: {
+    primary: {
+      main: "#FF4300",
+    },
+  },
+  typography: {
+    allVariants: {
+      fontFamily: 'IBM Plex Sans, Arial, sans-serif',
+      textTransform: "none",
+      wordBreak: "break-word",
+    },
+    h3: {
+      fontSize: "1.3rem",
+      fontWeight: 600
+    },
+    h4: {
+      fontSize: "1rem",
+      fontWeight: 600
+    },
+    h5: {
+      fontSize: "1.1rem",
+      fontWeight: 600,
+      lineHeight: "1.2rem"
+    },
+    h6: {
+      fontFamily: "Noto Sans",
+      fontSize: "0.9rem",
+      padding: "0 1rem",
+      fontWeight: 400,
+      lineHeight: "1rem",
+    },
+    h7: { 
+      fontWeight: 600, 
+      fontSize: "0.8rem", 
+      lineHeight: "0.5rem"
+    }
+  },
+});
```
LOTS of hardcoded values, there should be a few constants, like colors, that are re-used

id13 commented on src/features/comments/comment.js:
```diff
@@ -0,0 +1,97 @@
+  const replies = comment.data?.replies?.data?.children;
+  const date = new Date(comment.data.created_utc * 1000);
+  const [showReplies, setShowReplies] = useState(false);
+  const toddleReplies = () => {
```
👶 🤡

atonkopi replied:
Fixed🤡

id13 commented on src/features/comments/comment.js:
```diff
@@ -0,0 +1,97 @@
+import { relativeDays } from "../../components/helperFunctions";
+export const Comment = ({ comment }) => {
+  const replies = comment.data?.replies?.data?.children;
+  const date = new Date(comment.data.created_utc * 1000);
```
This logic of doing * 1000 should be a helper, like `date_from_reddit_timestamp`

id13 commented on src/features/comments/comment.js:
```diff
@@ -0,0 +1,97 @@
+            <Box
+              sx={{
+                cursor: replies ? "pointer" : "auto",
+                p: "0 1rem",
```
What's this ?

id13 commented on src/features/comments/comment.js:
```diff
@@ -0,0 +1,97 @@
+              onClick={toddleReplies}
+            >
+              <Box sx={{ width: "max-content" }}>
+                <Typography variant="h7" gutterBottom={true}>
```
I am spotting a lot of variant=h7, probably worth creating a typography component for that and name it for what variant=h7 stands for in your GUI

id13 commented on src/features/comments/comment.js:
```diff
@@ -0,0 +1,97 @@
+              >
+                {replies?.length && (
+                  <>
+                    <ForumRoundedIcon color="action" sx={{ pr: "0.3rem" }} />
```
what's this ? `pr: "0.3rem"`

id13 commented on src/features/comments/comment.js:
```diff
@@ -0,0 +1,97 @@
+      ></ListItemText>
+      {showReplies ? (
+        <ListItemText
+          sx={{ p: "0 1rem" }}
```
what's this ?

id13 commented on src/features/comments/comments.js:
```diff
@@ -0,0 +1,34 @@
+  if (loading || !postComments) {
+    return (
+      <div>
+        <Typography variant="h7">Comments are loading...</Typography>
```
Loading states

atonkopi replied:
Done

id13 commented on src/features/comments/commentsSlice.js:
```diff
@@ -0,0 +1,38 @@
+      };
+    },
+    [getComments.rejected]: (state) => {
+      state.loading = false;
```
probably want to handle errors ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+import { relativeDays } from "../../components/helperFunctions";
+
+export const Post = ({ selectedSubreddit, post, fullVersion = false }) => {
+  const date = new Date(post.data.created_utc * 1000);
```
DRY

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+      <Card
+        sx={{
+          width: "auto",
+          mb: "1rem",
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+          width: "auto",
+          mb: "1rem",
+          display: "flex",
+          pb: 0,
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+      >
+        <Box
+          sx={{
+            background: "#f5f5f5",
```
hardcoded color ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+        >
+          <IconButton
+            aria-label="thumb up"
+            sx={{ m: "0.3rem 0.3rem 0", p: "0.3rem" }}
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+          </Typography>
+          <IconButton
+            aria-label="thumb down"
+            sx={{ m: "0 0.3rem 0.3rem", p: "0.3rem" }}
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+            width: "100%",
+            display: "flex",
+            flexDirection: "column",
+            pb: 0,
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+            sx={{
+              display: "flex",
+              flexDirection: "column",
+              p: 0,
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+              sx={{ cursor: fullVersion ? "auto" : "pointer" }}
+            >
+              <CardHeader
+              sx={{pb: "0.5rem"}}
```
what's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+                  height="500rem"
+                  src={post.data.media.reddit_video.fallback_url}
+                  preload="auto"
+                  controls={true}
+                  autoPlay={true}
+                  loop={true}
```
```suggestion
                  controls
                  autoPlay
                  loop
```

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+                  {post.data.selftext}
+                </Typography>
+              )}
+              <CardActions sx={{ p: "1rem" }}>
```
What's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+                  size="small"
+                >
+                  <Button aria-label="view comments">
+                    <ChatBubbleIcon sx={{ pr: "0.3rem" }} />
```
What's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+                    </Typography>
+                    <Typography
+                      variant="h7"
+                      sx={{ display: { xs: "none", sm: "flex" }, p: "0.2rem" }}
```
What's this ?

id13 commented on src/features/posts/post.js:
```diff
@@ -0,0 +1,176 @@
+import ThumbDownOffAltIcon from "@mui/icons-material/ThumbDownOffAlt";
+import { relativeDays } from "../../components/helperFunctions";
+
+export const Post = ({ selectedSubreddit, post, fullVersion = false }) => {
```
This component can probably be broken into smaller ones, like the whole part with thumb up/down or the bottom bar with comments.

This could allow for an even better architecture:
- PostItem (to be displayed in posts list on subreddits)
- FullPost (to be displayed on full post page)
This could make the code easier to maintain

id13 commented on src/features/posts/postsSlice.js:
```diff
@@ -0,0 +1,67 @@
+  async (postId) => {
+    return fetch(`https://api.reddit.com/api/info/?id=t3_${postId}`)
+      .then((res) => res.json())
+      .then((data) => Object.assign({}, data.data.children[0]));
```
I am not sure what you are doing here, is data.data.children[0] not enough ? why Object.assign ?

atonkopi replied:
Removed Object.assign

id13 commented on src/features/posts/postsSlice.js:
```diff
@@ -0,0 +1,67 @@
+    },
+    [getPosts.fulfilled]: (state, action) => {
+      state.loading = false;
+      const { data: { after, children } } = action.payload // action.payload = { kind: "Listing", data: { after: "blabla", children: []}}
```
Is this comment still relevant ?

id13 commented on src/features/subReddits/subreddit.js:
```diff
@@ -0,0 +1,63 @@
+        onClick={onClickHandler}
+        disableRipple
+      >
+        <ListItemAvatar minwidth={"150rem"}>
```
rem are weird here

id13 commented on src/features/subReddits/subreddit.js:
```diff
@@ -0,0 +1,63 @@
+          <Avatar
+            src={
+              thumbnail ||
+              "https://external-preview.redd.it/iDdntscPf-nfWKqzHRGFmhVxZm4hZgaKe5oyFws-yzA.png?width=640&crop=smart&auto=webp&s=bfd318557bf2a5b3602367c9c4d9cd84d917ccd5"
```
Should be a constant somewhere, beware of hardcoding values like this

atonkopi replied:
fixed

id13 commented on src/features/subReddits/subreddit.js:
```diff
@@ -0,0 +1,63 @@
+          />
+        </ListItemAvatar>
+        <ListItemText>
+          <Typography variant="h4">{subreddit.data.display_name.charAt(0).toUpperCase()+subreddit.data.display_name.slice(1)}</Typography>
```
`subreddit.data.display_name.charAt(0).toUpperCase()+subreddit.data.display_name.slice(1)` should be a helper function to clean the name

atonkopi replied:
done

id13 commented on src/features/subReddits/subreddit.js:
```diff
@@ -0,0 +1,63 @@
+          <Typography variant="h4">{subreddit.data.display_name.charAt(0).toUpperCase()+subreddit.data.display_name.slice(1)}</Typography>
+            <Typography
+              variant="h7"
+              sx={{ color: "rgba(0, 0, 0, 0.54)", whiteSpace: "nowrap", display: {xs: "none", sm: "block"} }}
```
hardcoded color

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+                  border: "0.1rem solid rgba(255, 67, 0, 0.5)",
+                  borderRadius: 50,
+                  fontSize: "0.6rem",
+                  px: "0.3rem",
+                  py: "0.2rem",
```
what's this ?

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+              <Typography
+                variant="h9"
+                sx={{
+                  color: "#FF4300",
+                  border: "0.1rem solid rgba(255, 67, 0, 0.5)",
```
hardcoded colors that should be in themes

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+          {`ABOUT 
+          ${subredditInfo.display_name}
+          COMMUNITY`}
+          {subredditInfo.advertiser_category && (
+            <Box
+              sx={{ m: 0, p: "0 1rem", position: "absolute", right: 0, top: 0 }}
+            >
+              <Typography
+                variant="h9"
+                sx={{
+                  color: "#FF4300",
+                  border: "0.1rem solid rgba(255, 67, 0, 0.5)",
+                  borderRadius: 50,
+                  fontSize: "0.6rem",
+                  px: "0.3rem",
+                  py: "0.2rem",
+                  textTransform: "none",
+                }}
+              >
+                {subredditInfo.advertiser_category}
+              </Typography>
+            </Box>
+          )}
```
This advertiser category should be a UI component

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+          <Typography
+            variant="body2"
+            sx={{
+              p: "0.5rem 1rem  0",
```
what's this ?

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+              <Typography variant="h7">Online 🟢</Typography>
+            </ListItemText>
+          </ListItem>
+          <ListItem sx={{ pt: 0 }}>
```
what's this ?

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+            </ListItemText>
+          </ListItem>
+          <ListItem sx={{ pt: 0 }}>
+            <ListItemIcon sx={{ minWidth: "2rem" }}>
```
what's this ?

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+              <CakeIcon aria-label="date of creation" color="disabled" />
+            </ListItemIcon>
+            <ListItemText>
+              <Typography variant="h7" color="grey">
```
hardcoded color, should come from theme

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+            <ListItemText>
+              <Typography variant="h7" color="grey">
+                Created:{" "}
+                {created.toLocaleDateString("en-us", {
+                  year: "numeric",
+                  month: "short",
+                })}
```
Should be a helper function

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -0,0 +1,109 @@
+            </ListItemIcon>
+            <ListItemText>
+              <Typography variant="h7" color="grey">
+                Created:{" "}
```
```suggestion
                Created:&nbsp
```
If your future colleagues are nice, they will juste laugh at it

id13 replied:
More infos here 🤓 https://www.w3schools.com/html/html_entities.asp

id13 commented on src/features/subReddits/subredditsList.js:
```diff
@@ -0,0 +1,33 @@
+  if (loading) {
+    return (
+      <div>
+        <Typography variant="h7">Subreddits are loading...</Typography>
```
loading states

atonkopi replied:
done

atonkopi committed:
Update src/components/postPage.js

Co-authored-by: Jean-David Fiquet <jdfiquet2@gmail.com>

atonkopi committed:
Update src/components/postsListPage.js

Co-authored-by: Jean-David Fiquet <jdfiquet2@gmail.com>

atonkopi committed:
Revised comments and introduced changes

Regarding:

-helper functions
-loading states
-unnecessary styling

atonkopi committed:
Merge branch 'router' of github.com:a-dylean/reddit-client-app into router

atonkopi committed:
Changes to architecture of the code

- Created new Components to split big ones into smaller
- Modified helper functions, etc.

atonkopi committed:
Further improvements

-Added variables to theme;
-Removed hard-coded values;

atonkopi committed:
Fixed error in the Comment component

-handeled cases when there's no data loaded

atonkopi committed:
Fixed bug in SubredditInfo component

-Banner image (if exists) loads when component gets updated

id13 commented on src/components/layout.js:
```diff
@@ -1,39 +1,36 @@
         </Toolbar>
       </AppBar>
-      <main>{children}</main>
+      <Container sx={{ mt: "4rem" }}>
```
hardcoded style value should be standardized and put into theme

id13 commented on src/components/postsListPage.js:
```diff
@@ -35,48 +39,36 @@ const PostsListPage = () => {
           spacing={3.5}
```
hardcoded style value should be standardized and put into theme

id13 commented on src/components/search.js:
```diff
@@ -23,17 +22,15 @@ const ParentDiv = styled("div")(({ theme }) => ({
-  borderRadius: 50,
-  backgroundColor: "#fafafa",
-  color: "rgba(0, 0, 0, 0.54)",
+  borderRadius: 50, // theme.shape.borderRadius * 50,
```
hardcoded style value should be standardized and put into theme

id13 commented on src/components/search.js:
```diff
@@ -45,11 +42,17 @@ const StyledInputBase = styled(InputBase)(({ theme }) => ({
 }));
 
+const SuggestedSubreddits = styled(Card)(() => ({
+  boxShadow: "rgba(0, 0, 0, 0.35) 0px 5px 15px",
```
hardcoded style value should be standardized and put into theme

id13 commented on src/components/search.js:
```diff
@@ -68,47 +71,40 @@ export const Search = ({ hideSudreddit, showSubreddit }) => {
 
   const handleMouseEnter = () => {
     setShowSubreddits(true);
+    size.width < 400 && onFocusChange(false);
```
should it not be true ? when mouse enters, we are focusing, so onFocusChange should send new value as true

id13 commented on src/components/search.js:
```diff
@@ -68,47 +71,40 @@ export const Search = ({ hideSudreddit, showSubreddit }) => {
 
   const handleMouseLeave = () => {
     setShowSubreddits(false);
+    size.width < 400 && onFocusChange(true);
```
Same remark here

id13 commented on src/features/comments/comment.js:
```diff
@@ -1,96 +1,112 @@
+import { grey } from "@mui/material/colors";
+
+const CommentBox = styled(Box)(() => ({
+  padding: "0 1rem",
```
hardcoded style value should be standardized and put into theme

id13 commented on src/features/posts/postItem.js:
```diff
@@ -0,0 +1,131 @@
+              component="img"
+              src={post.data.url}
+              alt="post image"
+              height="400rem"
```
hardcoded style value should be standardized and put into theme, and I don't see why rem is required here

id13 commented on src/features/posts/postItem.js:
```diff
@@ -0,0 +1,131 @@
+            {post.data.media?.reddit_video && (
+              <CardMedia
+                component="video"
+                height="500rem"
```
hardcoded style value should be standardized and put into theme, same remark with rem

id13 commented on src/features/posts/postItem.js:
```diff
@@ -0,0 +1,131 @@
+                sx={{ backgroundColor: "black" }}
+              />
+            )}
+            {post.data.selftext.length >= 300 ? (
```
should be specified into constants or theme as a text length breakpoint

id13 commented on src/features/posts/sideBar.js:
```diff
@@ -0,0 +1,36 @@
+  return (
+    <Box
+      sx={{
+        background: grey[100],
```
hardcoded style value should be standardized and put into theme

id13 commented on src/features/posts/sideBar.js:
```diff
@@ -0,0 +1,36 @@
+    >
+      <IconButton
+        aria-label="thumb up"
+        sx={{ m: "0.3rem 0.3rem 0", p: "0.3rem" }}
```
hardcoded style value should be standardized and put into theme

id13 commented on src/features/posts/sideBar.js:
```diff
@@ -0,0 +1,36 @@
+      </Typography>
+      <IconButton
+        aria-label="thumb down"
+        sx={{ m: "0 0.3rem 0.3rem", p: "0.3rem" }}
```
hardcoded style value should be standardized and put into theme

id13 commented on src/features/subReddits/accountsInfo.js:
```diff
@@ -0,0 +1,16 @@
+import { Typography,   ListItemText } from "@mui/material";
+const AccountsInfo = ({num, text}) => {
+  return (
+            <ListItemText sx={{ display: "flex", justifyContent: "center" }}>
+              <Typography variant="h5">
+                {num}
+                <br />
+              </Typography>
+              <Typography variant="subtitle2">{text}</Typography>
+            </ListItemText>
+                  
```
Need to be formatted, you should have an auto-format installed :)

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -2,105 +2,103 @@ import React, { useEffect } from "react";
+
+const AdvertiserCategory = styled("p")(({ theme }) => ({
+  color: theme.palette.primary.main,
+  lineHeight: "1.5rem",
+  border: "0.1rem solid rgba(255, 67, 0, 0.5)",
+  borderRadius: "5rem",
+  fontSize: "0.7rem",
```
hardcoded style value should be standardized and put into theme

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -2,105 +2,103 @@ import React, { useEffect } from "react";
+  borderRadius: "5rem",
+  fontSize: "0.7rem",
+  textTransform: "none",
+  padding: theme.spacing(0, 0.5),
```
hardcoded style value should be standardized and put into theme, although you use the theme, as a base, still hardcoded

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -2,105 +2,103 @@ import React, { useEffect } from "react";
+  padding: theme.spacing(0.5, 2),
+  margin: 0,
+  textAlign: "justify",
+  fontSize: "0.85rem",
```
hardcoded style value should be standardized and put into theme

id13 commented on src/features/subReddits/subredditInfo.js:
```diff
@@ -2,105 +2,103 @@ import React, { useEffect } from "react";
-            </ListItemText>
-          </ListItem>
-        </CardContent>
+          height="100rem"
```
hardcoded style value should be standardized and put into theme

id13 reviewed (changes requested):
Congratulations on improving the code 👍

atonkopi committed:
Render text in markdown

atonkopi committed:
Merge branch 'main' of github.com:a-dylean/reddit-client-app into router
